scarlet/lite/frame.py

```diff
@@ -0,0 +1,258 @@
+import numpy as np
+
+
+class CartesianFrame:
+    """A grid of X and Y values contained in a bbox
+    """
+
+    def __init__(self, bbox):
+        # Store the new bounding box
+        self._bbox = bbox
+        # Get the range of x and y
+        yi, xi = bbox.start[-2:]
+        yf, xf = bbox.stop[-2:]
+        height, width = bbox.shape[-2:]
+        y = np.linspace(yi, yf - 1, height)
+        x = np.linspace(xi, xf - 1, width)
+        # Create the grid used to create the image of the frame
+        self._X, self._Y = np.meshgrid(x, y)
```

Comment:
Seems a bit memory-inefficient to always store this grid for every image, but I suppose it's unavoidable if you're not going to write out loops explicitly.

Reply:
Yes, this was a part of the design that I struggled with. I tried with both caching and not caching, and the `numpy.meshgrid` was expensive enough that it made sense to store the entire grid once for the entire blend and slice those coordinates appropriately when models are generated for individual sources (which is why the `frame` is passed around in the `ParametericComponent` models.